R/plot_utils.R

```diff
@@ -163,25 +163,18 @@ cxSupportedLineFit <- function(linefit) {
 }
 
 rgbaConversion <- function(color, alpha = 0.5){
-    rgbaVal <- paste(c(col2rgb(color), alpha), collapse = ",")
-    rgbaStr <- paste0("rgba(", rgbaVal, ")")
-    return(rgbaStr)
-}
-
-getCxPlotDecorations <- function(decorations, color, width, x, y) {
-    line <- list(color = color,
-                 width = width)
-    if (!missing(x) && !missing(y)) {
-        line <- append(line, list(x  = x,
-                                  x2 = y,
-                                  y  = x,
-                                  y2 = y))
-    } else if (!missing(x)) {
-        line <- append(line, list(x = x))
-    } else if (!missing(y)) {
-        line <- append(line, list(y = y))
+    rgbastr <- NULL
+    if (!is.character(color)) {
+        rgbastr <- "invalid value"
     }
-
-    decorations <- list(line = append(decorations$line, list(line)))
-    return(decorations)
+    tryCatch({
+        rgbaVal <- paste(c(col2rgb(color), alpha), collapse = ",")
+        rgbastr <- paste0("rgba(", rgbaVal, ")")
+    }, error = function(e) {
+        warning(paste("Ïnvalid color specified",color))
+    })
+    if (is.null(rgbastr)) {
+        rgbastr <- "invalid value"
+    }
+    return(rgbastr)
```

Comment:
just type **rgbastr** and it will return automatically

Reply:
done